src/V3LinkDot.cpp

```diff
@@ -3267,6 +3275,7 @@ class LinkDotResolveVisitor final : public VNVisitor {
     }
     void visit(AstClass* nodep) override {
         nodep->user3SetOnce();
+        m_classp = nodep;
         UINFO(5, "   " << nodep << endl);
         checkNoDot(nodep);
         VL_RESTORER(m_curSymp);
```

Comment:
```suggestion
        UINFO(5, "   " << nodep << endl);
        checkNoDot(nodep);
        VL_RESTORER(m_classp);
        VL_RESTORER(m_curSymp);
        m_classp = nodep;
```

Otherwise m_classp stays set after class ends.